cmake/thirdparty/get_legion.cmake

```diff
@@ -0,0 +1,104 @@
+#=============================================================================
+# Copyright 2022 NVIDIA Corporation
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#=============================================================================
+
+include(${CMAKE_CURRENT_SOURCE_DIR}/cmake/Modules/legion_helpers.cmake)
+
+function(find_or_configure_legion)
+  set(oneValueArgs VERSION REPOSITORY PINNED_TAG EXCLUDE_FROM_ALL)
+  cmake_parse_arguments(PKG "${options}" "${oneValueArgs}" "${multiValueArgs}" ${ARGN})
+
+  set(Legion_CUDA_ARCH "")
+  if(Legion_USE_CUDA)
+    set(Legion_CUDA_ARCH ${CMAKE_CUDA_ARCHITECTURES})
+    list(TRANSFORM Legion_CUDA_ARCH REPLACE "-real" "")
+    list(TRANSFORM Legion_CUDA_ARCH REPLACE "-virtual" "")
+    list(JOIN Legion_CUDA_ARCH "," Legion_CUDA_ARCH)
+  endif()
+
+  # Detect the presence of LIBRARY_PATH envvar so we can set
+  # `CMAKE_LIBRARY_PATH` for Legion's FindCUDA.cmake calls.
+  set(_lib_path "${CMAKE_LIBRARY_PATH}")
+  if(DEFINED ENV{LIBRARY_PATH})
+    list(APPEND _lib_path "$ENV{LIBRARY_PATH}")
+  endif()
+
+  set(FIND_PKG_ARGS      ${PKG_VERSION}
+      GLOBAL_TARGETS     Legion::Realm
+                         Legion::Regent
+                         Legion::Legion
+                         Legion::RealmRuntime
+                         Legion::LegionRuntime
+      BUILD_EXPORT_SET   legate-core-exports
+      INSTALL_EXPORT_SET legate-core-exports)
+
+  # First try to find Legion via find_package()
+  # so the `Legion_USE_*` variables are visible
+  rapids_find_package(Legion ${FIND_PKG_ARGS} QUIET)
```

Comment:
just out of curiosity, what extra features does rapids_find_package provide over native find_package? AFAICT from online docs (which, kudos to rapids for having extensive online docs of their cmake libraries) this basically helps in adding an imported library to an export set?

Reply:
LGTM. I can't mark resolved.